src/Types/ReceiveName.cs

```diff
@@ -3,4 +3,4 @@ namespace Concordium.Sdk.Types;
 /// <summary>
 /// A receive name of the contract function called. Expected format: "&lt;contract_name&gt;.&lt;func_name&gt;".
 /// </summary>
-public readonly record struct ReceiveName(string Receive);
+public sealed record ReceiveName(string Receive);
```

Comment:
Not sure what you mean - can you give some more details?

Reply:
I have added public constructors with validations and written test. Please review these.

Reply:
I have removed InitName in favor of only using ContractName, since this is the mapping done in Rust

https://github.com/Concordium/concordium-rust-sdk/blob/9445ecafd148d856fbb81da14c1c013ec4637619/src/v2/conversions.rs#L169


is it really true contract name always has a `init_` prefix? Why does it have this?